src/components/SignInForm/SignInForm.jsx

```diff
@@ -0,0 +1,168 @@
+    const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+    if (!emailRegex.test(email)) {
+      setErrors((prevErrors) => ({
+        ...prevErrors,
+        email: { valid: false, message: "Не правильна адреса" },
+      }));
+      return false;
+    }
+    setErrors((prevErrors) => ({
+      ...prevErrors,
+      email: { valid: true },
+    }));
+    return true;
+  };
+
+  const validatePassword = (password) => {
+    if (password.length < 8) {
+      setErrors((prevErrors) => ({
+        ...prevErrors,
+        password: {
+          valid: false,
+          message:
+            "Не меньше 8 символів",
+        },
+      }));
+      return false;
+    }
+    setErrors((prevErrors) => ({
+      ...prevErrors,
+      password: { valid: true },
+    }));
+    return true;
+  };
+
+  const handleSignIn = async (event) => {
+    event.preventDefault();
+    setIsSubmitting(true);
+
+    if (validateEmail(email) && validatePassword(password)) {
+      try {
+        const response = await fetch("/api/v1/user/login", {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify({ email, password }),
+        });
+        const data = await response.json();
+
+        if (response.ok) {
+          authContext.loginCurrentUser(data.jwt);
+          setEmail("");
+          setPassword("");
+          setErrors({});
+        } else {
+          const errorMessage =
+            data.message ||
+            "Такого користувача не існує, перевірте правильність введених даних";
+          throw new Error(errorMessage);
+        }
+      } catch (error) {
+        console.error("Failed:", error);
+      }
+    }
+    setIsSubmitting(false);
+  };
+
+  return (
+    <div className={styles.signInContainer}>
+      <div>
+        <img src="/images/sign-in.png" alt="food" className={styles.imgFood} />
+      </div>
+      <div className={styles.formContainer}>
+        <h2>Вхід</h2>
+        <form onSubmit={handleSignIn}>
+          <div className={styles.emailContainer}>
+            <label htmlFor="email">E-mail</label>
+            <br />
+            <input
+              style={{
+                border: errors.email?.valid === false ? "1px solid #E02D3C" : "",
+              }}
+              type="email"
+              id="email"
+              name="email"
+              required
+              value={email}
+              onChange={handleEmailChange}
+              onBlur={() => validateEmail(email)}
+              autoFocus
+            />
+            {errors.email?.valid === false && (
+              <p className={styles.incorrectInputMessage}>{errors.email.message}</p>
+            )}
+          </div>
+          <div className={styles.passwordContainer}>
+            <label htmlFor="password">Пароль</label>
+            <input
+              style={{
+                border: errors.password?.valid === false ? "1px solid #E02D3C" : "",
```

Comment:
better this way: 
```js
style={{
    border: errors.password?.valid ? "" : "1px solid #E02D3C"
}}
```

Reply:
there were difficulties here - the valid field(in object errors) does not exist for some time, but according to the design requirement, the input should be green until it passes validation, so I left the previous option

Reply:
yeah, you are right. 
your way is more reliable. 

the one i proposed above won't work properly, until we introduce one more condition in it:
`!errors.email || errors.email?.valid ? '' : '1px solid #E02D3C'`

but yes, 
let's keep your implementation, it seems to be better.

Reply:
I've got it